### ISSUE TITLE
Cannot write into canvas on LM with graphics tablet

### ISSUE THREAD

**Issue by MStarha** [xournalpp/xournalpp#2932]
I am running Xournal++ 1.0.20 on my Linux Mint 20.1 and I am having a problem with my Waltop graphics tablet.

In settings the tablet is recognized and is assigned default configuration. However, I cannot write with my pen. I can select in the menu but nothing happens on the canvas. Doing so with my mouse works normally.

Even if I set the tablet to be used as "pen" in settings, the behavior is still the same. I also tried various combinations without success.

I did not have this issue with Xournal, but since it is not maintained I wanted to switch to Xournal++.

Thank you for help and your time.

**Comment by MStarha** [comment]
Here is the screenshot. This is the default configuration without changing anything.

![xournalpp_settings](https://user-images.githubusercontent.com/59487310/110667291-669eeb80-81ca-11eb-854f-615c98cca622.png)

**Comment by MStarha** [comment]
In the original post I stated that it did not work. But I have tried it again with the new version and writing now works as expected. But I still cannot use eraser despite having set it to behave as eraser in the "Input System" and "Stylus" settings panel.

**Comment by rolandlo** [comment]
Maybe the eraser is reported as a different button. It could also be that your stylus is configured as a mouse. Try making some changes in the "Mouse" and "Stylus" settings panels and see if the eraser button starts to do something.

**Comment by MStarha** [comment]
I tried again various configuration in the "Mouse" tab, tried setting the tablet as mouse, as eraser,... nothing.

I had some success by setting the buttons in "Stylus" tab as eraser. The cursor change to eraser but it did not erase anything, just floated above the pen strokes.

**Comment by MStarha** [comment]
Just to clarify: I do not have touch screen, just a regular graphics tablet (pen, and sensitive pad).

The "Merge button events..." did in fact work. The eraser is now active when I press a button on the pen and does not matter, whether I touch with the pen tip or just hover above the surface.

Anyway, it is set up as I wanted. It required a bit of tinkering with the settings, though. I can now close this issue.

Thank you.

**Comment by rolandlo** [comment]
Can you just summarize for other users which settings had to be changed for your stylus and eraser to work?

**Comment by MStarha** [comment]
In "Input system" settings panel the tablet name is followed by "(touchscreen)" and is set up as touchscreen. This setting has to be changed to "Pen". Bellow, in "Options", the "Merge button events with stylus tip events" checkbox must be tick in order for the eraser to work. Then in "Stylus" settings panel I set button 1 to be interpreted as eraser.

Now, when I try to write, the response is as expected. A stroke appears on the canvas and the width depends on the pressure applied to the pen. When I press button 1, the tool changes to eraser and is active the whole time the button is pressed. Applying pressure to the tip is irrelevant despite having ticked "Merge button events with tip events", which to me suggests, that the eraser should be active when the button is pressed and pressure is applied to the tip as well.

These changes help only when I am running the latest nightly build (1.1.0). When I tried it with the latest stable version (1.0.20), nothing of the mentioned above helps. I cannot write, erase,... with my tablet and pen.
